pyasdf/tags/wcs/wcs.py

```diff
@@ -4,99 +4,229 @@
 from __future__ import absolute_import, division, unicode_literals, print_function
 
 
+import six
+
+
 from ...asdftypes import AsdfType
 from ... import yamlutil
 
 
-class GWCSAxis(object):
-    def __init__(self, **kwargs):
-        self.__dict__.update(kwargs)
+_REQUIRES = ['gwcs', 'astropy']
 
 
-class GWCSStep(object):
-    def __init__(self, name, axes, transform):
-        self.name = name
-        self.axes = axes
-        self.transform = transform
+class WCSType(AsdfType):
+    name = "wcs/wcs"
+    requires = _REQUIRES
+    types = ['gwcs.WCS']
 
+    @classmethod
+    def from_tree(cls, node, ctx):
+        import gwcs
 
-class GWCS(object):
-    def __init__(self, steps):
-        self.steps = steps
-        self.transform = steps[0].transform
-        for step in steps[1:-1]:
-            self.transform = self.transform | step.transform
+        steps = [(x['frame'], x.get('transform')) for x in node['steps']]
+        name = node['name']
 
-        if self.steps[-1].transform is not None:
-            raise ValueError("Last WCS step must not have a transform defined")
+        return gwcs.WCS(steps, name=name)
 
+    @classmethod
+    def to_tree(cls, gwcs, ctx):
+        frames = gwcs.available_frames
+        steps = []
+        for i in range(len(frames) - 1):
+            transform = gwcs.get_transform(frames[i], frames[i + 1])
+            steps.append(StepType({'frame': frames[i], 'transform': transform}))
```

Comment:
This will not write to file the entire frame object. The reason is that `available_frames` is a list of the frame names.
To get the frame object use 
```
frame = getattr(gwcs, frame_name)
if frame is None:
    frame = frame_name
```
 Frames can be strings or instances of `CoordinateFrame`. So this will return `None` if a frame is just a string.